### ISSUE TITLE
Rustdoc does not add tooltips for reference links

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#89246]
In general, links generated by rustdoc do have a tooltip.

E.g. something like
```rs
/// Talking about [`Result`][std::fmt::Result] here!
pub mod m {}
```
will come with a tooltip

![Screenshot_20210906_223756](https://user-images.githubusercontent.com/3986214/132259375-864c8d00-99e8-4d82-abc2-feddc472b605.png)

Whereas (unfortunately) the following alternative
```rs
/// Talking about [`Result`] here!
/// 
/// [`Result`]: std::fmt::Result
pub mod m {}
```
creates almost the same documentation, just lacking the tooltip.

_Originally posted by @steffahn in https://github.com/rust-lang/rust/pull/88343#r703066344_

**Comment by GuillaumeGomez** [comment]
I'll fix it tomorrow.

**Comment by GuillaumeGomez** [comment]
So, I think here that this is a markdown issue rather than a rustdoc issue. For example, there isn't a title there either:

```rust
/// [`Result`](std::result::Result]
```

Do you want us to add the title if there isn't one which will be the path to the type?

**Comment by GuillaumeGomez** [comment]
It was fixed in #108025.
